### PR TITLE
feat(api): endpoints to get and update key dates

### PR DESCRIPTION
**Pull request by chrisdalykinandcarta**
## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: apps/api/src/server/swagger.js
Comment (beejjacobs) [inline apps/api/src/server/swagger.js:1]: Just need the corresponding swagger-output update

**Review by beejjacobs — CHANGES_REQUESTED** [review]
Few changes, but otherwise looks good!

File: apps/api/src/server/applications/key-dates/key-dates.mapper.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+export const mapRequestToKeyDates = ({
+	preApplication,
+	preExamination,
+	examination,
+	recommendation,
+	decision,
+	postDecision,
+	withdrawal
+}) => {
+	const allKeyDates = {
```
[the remark this answers is not in the document]
Comment (chrisdalykinandcarta) [reply apps/api/src/server/applications/key-dates/key-dates.mapper.js:29]: The request will come structured with top-level properties like preApplication and preExamination, but the database stores them all as flat properties.

The top-level organisation is for the benefit of the consumer.

File: apps/api/src/server/applications/key-dates/key-dates.mapper.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		...preApplication,
+		...preExamination,
+		...examination,
+		...recommendation,
+		...decision,
+		...postDecision,
+		...withdrawal
+	};
+
+	return pick(allKeyDates, allDateKeys);
```
[the remark this answers is not in the document]
Comment (chrisdalykinandcarta) [reply apps/api/src/server/applications/key-dates/key-dates.mapper.js:39]: Haha you're right, good suggestion!

File: apps/api/src/server/swagger.js
Comment (chrisdalykinandcarta) [reply apps/api/src/server/swagger.js:1]: How is this generated?

File: apps/api/src/server/utils/mapping/map-date-to-unix-timestamp.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+/**
+ *
+ * @param {Date} date
+ * @returns {number}
+ */
+export const mapDateToUnixTimestamp = (date) => Math.floor(date.getTime() / 1000);
```
Comment (lutamKC) [inline apps/api/src/server/utils/mapping/map-date-to-unix-timestamp.js:6]: 👍

**Review by lutamKC — APPROVED** [review]

**Review by beejjacobs — APPROVED** [review]
LGTM!

**Pull request merged**